### PR TITLE
feat(config): create config package

### PR DESCRIPTION
**Pull request by dk1a**
similar to https://github.com/latticexyz/mud/pull/584 but config has no solidity/forge

### CODE REVIEW

File: packages/config/.gitignore [hunk 1]
```diff
@@ -0,0 +1,12 @@
+cache
+abi
+out
+dist
+types/ethers-contracts
```
Comment (alvrs) [inline packages/config/.gitignore:5]: abi, out, ethers-contracts is just needed in solidity packages

File: packages/config/vite.config.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  build: {
+    lib: {
+      entry: "src/index.ts",
+      formats: ["cjs", "es"],
+    },
+    outDir: "dist",
+    minify: false,
+    sourcemap: true,
+  },
+  test: {
```
Comment (roninjin10) [inline packages/config/vite.config.ts:18]:
```suggestion
  /**
   * @see https://vitest.dev/config/
   */
  test: {
```

File: packages/config/.nvmrc [hunk 3]
```diff
@@ -0,0 +1 @@
+18
```
Comment (alvrs) [inline packages/config/.nvmrc:1]: Don't think we need a local .nvmrc, @holic moved this to a root one recently

File: packages/config/package.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  "main": "dist/index.js",
+  "module": "dist/index.mjs",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/latticexyz/mud.git",
+    "directory": "packages/config"
+  },
+  "scripts": {
+    "prepare": "yarn build",
+    "test": "echo 'TODO tests'",
```
Comment (roninjin10) [inline packages/config/package.json:17]:
```suggestion
    "test": "vitest",
```
Could just use vitest (loooks like you were headed that way)

**Review by alvrs — APPROVED** [review]
lgtm, just tiny nits

File: packages/config/yarn.lock [hunk 5]
```diff
@@ -0,0 +1,645 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (roninjin10) [inline packages/config/yarn.lock:1]: Wierd yarn generated this

Comment (roninjin10) [reply packages/config/yarn.lock:1]: You will want to remove this and run yarn again from root btw

Comment (dk1a) [reply packages/config/yarn.lock:1]: oh right I forgot to add the package to "workspaces"

File: packages/config/package.json (hunk 4 above)
Comment (dk1a) [reply packages/config/package.json:17]: can't do that rn - I have no actual tests and vitest errors because of it, wouldn't want to break our CI

Comment (dk1a) [reply packages/config/package.json:17]: but yeah the plan is to do `vitest` when I make tests

**Review by holic — APPROVED** [review]

**Pull request merged**